### ISSUE TITLE
DPO deepspeed OOM with optimizer

### ISSUE THREAD

**Issue by Jack000** [huggingface/trl#959]
I was trying to use dpo.py with deepspeed and kept getting OOM at this line:

[       model, *_ = deepspeed.initialize(model=model, config=config_kwargs)](https://github.com/huggingface/trl/blob/6ff0fac2c118250e1a14a0a793e96c5c3f654885/trl/trainer/dpo_trainer.py#L324C2-L324C76)

this seems to just use the base model's config, which has optimizer settings, it then tries to initialize the optimizer leading to OOM. The reference model is static (and is set to eval mode on the very next line) so the optimizer is not needed.

I got it working by copying the config and setting kwargs["optimizer"] = {"type": None }

**Comment by hanyullai** [comment]
Hi, I also run into similar issue. Could you show me the code how to disable the optimizer states of the ref model. I try the following but still not working.
```python
copy_config = config_kwargs.copy()
copy_config["optimizer"] = {"type": None}
model, *_ = deepspeed.initialize(model=model, config=copy_config)
model.eval()
return model
```

**Comment by lvwerra** [comment]
This sounds good to me. Maybe @lewtun and @kashif can confirm. Would you mind opening a PR? The config copy should be implemented since #957.

**Comment by lewtun** [comment]
Nice catch @Jack000 ! I agree with the proposal and would be happy to review a PR :)

**Comment by lewtun** [comment]
GitHub bot begone!
